remote/bitbucketserver/internal/client.go

```diff
@@ -214,3 +250,142 @@ func (c *Client) paginatedRepos(start int) ([]*Repo, error) {
 	}
 	return repoResponse.Values, nil
 }
+
+func Filter(vs []string, f func(string) bool) []string {
+	vsf := make([]string, 0)
+	for _, v := range vs {
+		if f(v) {
+			vsf = append(vsf, v)
+		}
+	}
+	return vsf
+}
+
+//TODO: find a clean way of doing these next two methods- bitbucket server hooks only support 20 cb hooks
+func arrayToHookSettings(hooks []string) HookSettings {
```

Comment:
Can you provide a bit of context? The code looks a bit unorthodox, but I get the sense that this is because the Stash API is not user-friendly. So I think the code is probably fine to merge (thanks!) but I just want to understand a bit better.